### ISSUE TITLE
Missing matching element for `aria-controls` value

### ISSUE THREAD

**Issue by raebot** [jupyter-book/myst-theme#808]
A WCAG audit of pages generated with book theme is reporting an invalid value with the `aria-controls` attributes on the page. If I understand the [documentation on that attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-controls), the value is intended to match the `id` attribute of another element in the document. The search button element at the top right of the page is an example where there is not a matching element as far as I can see. 

```
<button type="button" aria-haspopup="dialog" aria-expanded="false" aria-controls="radix-:R75cp:" data-state="closed" class="myst-search-bar flex items-center h-10 aspect-square sm:w-64 text-left text-gray-600 dark:text-gray-300 border border-gray-300 dark:border-gray-600 rounded-lg bg-gray-50 dark:bg-gray-700 myst-search-bar-disabled hover:ring-blue-500 dark:hover:ring-blue-500 hover:border-blue-500 dark:hover:border-blue-500">...</button>
```

This is the error I see when running the `axe-selenium-python` package against the JupyterBook pages. I was not able to reproduce this with axe DevTools in Chrome.

```
- 🛑‼️  ARIA attributes must conform to valid values (https://dequeuniversity.com/rules/axe/3.1/aria-valid-attr-value?application=axeAPI):

  Fix all of the following:
  Invalid ARIA attribute value: aria-controls="radix-:R75cp:"

  HTML fragment: <button type="button" aria-haspopup="dialog" aria-expanded="false" aria-controls="radix-:R75cp:" dat
```

Thanks for looking into this!

**Comment by choldgraf** [comment]
I asked Claude to poke around at this because I find ui and accessibility stuff to be quite hard to debug, but also important to fix! Here's what I found so far, I think we could fix this as long as we are okay removing aria-controls. Here is some context on that...

It seems like the answer is that this is a radix bug, where the aria-controls is referring to something that doesn't actually exist while the search dialogue is closed. there's actually a bug about this here:

- https://github.com/radix-ui/primitives/issues/3560

one option would be to just not use "aria-controls" here at all, since it doesn't seem to be super well supported anyway. If we did that, it should remove this error.

I looked into aria-controls and it generally doesn't seem to be super well supported anyway. Here's a reference suggesting it doesn't have great support across browsers and readers, and I think the button already has other aria attributes attached to it

https://a11ysupport.io/tech/aria/aria-controls_attribute

And here is a kind of ridiculous blog post from somebody that really doesn't like it 😅

https://heydonworks.com/article/aria-controls-is-poop/

So I think my vote would be to remove it

**Comment by erikamov** [comment]
Any updates for this issue?

**Comment by choldgraf** [comment]
Not that I know of - if you're interested in giving a shot at a solution I'd be happy to review. Right now I think there's some clarity on what needs to be done, but we're waiting for somebody w/ the bandwidth to try fixing it
